Library/Homebrew/formulary.rb

```diff
@@ -123,7 +123,17 @@ def initialize(bottle_name)
     end
 
     def get_formula(spec, alias_path: nil)
-      formula = super
+      bottle_version = Utils::Bottles.resolve_version @bottle_filename
+      path = "#{name}/#{bottle_version}/.brew/#{name}.rb"
+      contents = Utils.popen_read "tar", "-xO", "-f", @bottle_filename,
+        "--strip-components", "3", path, err: :close
+      formula = if $CHILD_STATUS.success?
+        Formulary.from_contents name, HOMEBREW_CELLAR/path, contents, spec
+      else
+        opoo "This bottle does not contain a formula: #{path}"
```

Comment:
I think this should probably just fail rather than have inconsistent behaviour. If you're loading bottles more than a year old you're going to have other problems.

Reply:
Agreed. Done.